x/evm/keeper/keeper.go

```diff
@@ -181,9 +181,15 @@ func (k *Keeper) GetVMBlockContext(ctx sdk.Context, gp core.GasPool) (*vm.BlockC
 		return nil, err
 	}
 	rh := common.BytesToHash(r)
+
+	txfer := core.Transfer
```

Comment:
this can be changed to the following to account for internal transactions' transfer as well
```
func(db vm.StateDB, sender, recipient common.Address, amount *big.Int) {
    if IsPayablePrecompile(recipient) {
        state.TransferWithoutEvents(db, sender, recipient, amount)
    } else {
        core.Transfer(db, sender, recipient, amount)
    }
}
```